tools/autograd/gen_variable_type.py

```diff
@@ -847,12 +855,33 @@ def emit_increment_version():
             return []
         return ['increment_version({});'.format(arg['name']) for arg in differentiable_outputs]
 
+    def check_type_for_capture(arg_simple_type):
+        return arg_simple_type in ['Tensor', 'Scalar']
+
+    def record_function_inputs_env():
+        if declaration['api_name'] == '_cudnn_rnn':
+            print(declaration)
+        inputs_env = {}
+        args_to_capture = [
+            arg for arg in declaration['arguments']
+            if check_type_for_capture(arg['simple_type'])]
+        inputs_env['capture_inputs'] = ', '.join(
+            ['&{}'.format(arg['name']) for arg in args_to_capture])
+        inputs_env['push_inputs'] = '\n'.join(
+            ['jit::push(inputs, {});'.format(arg['name']) for arg in args_to_capture])
+        return inputs_env, args_to_capture
```

Comment:
What is up with signature here? Why can't it just return the string for the capture argument?

Reply:
it's more generic, in case you'd want to get access to args used, but we can return just a string for sure